geps/gep-1911.md

```diff
@@ -0,0 +1,190 @@
+  HTTP2     *HTTP2PortProtocol    `json:"http2"`
+  HTTP1     *HTTP1PortProtocol    `json:"http1"`
+  GRPC      *GRPCPortProtocol     `json:"grpc"`
+  WebSocket *WebsocketPortProtocol`json:"websocket"`
+}
+
+type HTTP2PortProtocol struct {
+  Cleartext bool // h2c when set to true
+
+  // Future knobs can go here
+}
+
+type HTTPPortProtocol struct {
+  // Future knobs can go here
+}
+type GRPCPortProtocol struct {
+  // Future knobs can go here
+}
+type WebsocketPortProtocol struct {
+  // Future knobs can go here
+}
+```
+
+### YAML API
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: BackendProtocol
+metadata:
+  name: some-service
+  namespace: some-namespace
+spec:
+  ports:
+  - port: 80
+    http2: 
+      encrypted: false # enable h2c
+  - port: 80
+    http1: {}          # We also listen to http1 on port 80
+  - port: 9090
+    grpc: {}
+  - port: 8080
+    websocket: {}      # enable websocket support on this port
+  - port: 7070
+    http1: {}
+  targetRef:
+    group: "" # corev1
+    kind: Service
+    name: some-service
+```
+
+### Default Protocols
+
+If a backend protocol isn't specified an implementation MAY infer the backend protocol through it's own means.
+
+### Incompatible Protocols
+
+If a Route attached to a Gateway is not able to support sending traffic to the backend using a specific protocol 
+then it must set it's condition to `Accepted` `False`.
+
+## Alternatives
+
+### Meta-resource for each protocol
+
+Rather than bundle protocol details into a single resource an alternative would be to create distinct meta resources.
+ie. `HTTP2Backend`, `GPRCBackend`, `WebsocketBackend`.
+
+Although this has the advantage of seperate it doesn't allow
+
+### Adding properties on Gateway Route Objects
+
+From [GEP-1282](geps/gep-1282.md#tldr):
+> some types of configuration requested by users are more about defining functionality that describes capabilities of the backend more than the route you take to get to the backend.
+
+Backend protocol is specifying capabilities. This configuration is less about routing.
+
+### Extending Kubernetes Service
+
+This is considered untenable due to the 'the turnaround time for those changes can be years.' ([ref-1282](geps/gep-1282.md#non-goals):
+
+### AppProtocol
+
+From [GEP-1282](geps/gep-1282.md#non-goals):
+> v1.Service’s appProtocol field is not fit for purpose, because it is defined as accepting values either from the IANA Service Name registry, or domain-prefixed values and we need more flexibility than that. 
+
+See further discussion [here](https://github.com/kubernetes-sigs/gateway-api/discussions/1244#discussioncomment-3151197)
+
+An option would be to have a specific vendor prefix but unfortunately this limits the protocol to one per port.
+
+### Unstructured Data/Special Values
+
+Unstructured data refers to using labels and annotations.
+
+From [GEP-1282](geps/gep-1282.md#non-goals):
+> these are very sticky and hard to get rid of once you start using them.
```

Comment:
Annotations are also not self-documenting, there is no `kubectl explain` for annotations.

Reply:
added this to the section